assemble/src/main/scripts/generate-download-script.sh

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will generate a DEPENDENCIES listing of packaged dependencies
+
+in=target/dependencies.raw.txt
+out=target/download-dependencies
+
+cat >"$out" <<'EOF'
+#! /usr/bin/env bash
+# This script downloads the following jars, identified by their maven
+# coordinates, using the maven-dependency-plugin.
+#
+# DISCLAIMER: This is only one possible way to download a set of dependencies
```

Comment:
It could be better worded, yes, but I think it's important to communicate to users that this is not guaranteed to be the best option for their environment, and it may not even download the right jars they need. It is still something they may need to figure out how to do.... just like the classpaths in our configuration templates are not necessarily suitable for them (and in fact, they should never be used as is... because wildcard classpaths are a bad idea for security).

I don't know the best way to communicate their responsibility for their environment and demonstrate the utility of this as a convenience mechanism for accomplishing the task.

In a way, I thought this might be one way (in addition to the POM) of declaring what dependencies they'll need, but certainly am willing to phrase it better if you have a suggestion to do that. The maven site also generates a listing of necessary dependencies (if we were to build a maven site), so that's another possible way of communicating the dependency requirements. We could also append a list to the README. The binary artifacts also already include a DEPENDENCIES file which includes license information and dependency version information.